### ISSUE TITLE
stat/distuv: add getParameters() and parameters() calls to distributions missing them

### ISSUE THREAD

**Issue by NMRobert** [gonum/gonum#568]
Hey,

Some, but not all of the univariate distributions in stat/distuv have a method for getting/setting their parameters. I'd like to add these methods to at least some of the outstanding distributions, and perhaps standardize the method naming a bit.

The methods in question look like these: 
```
// setParameters modifies the parameters of the distribution.
func (n *Normal) setParameters(p []Parameter) {
	if len(p) != n.NumParameters() {
		panic("normal: incorrect number of parameters to set")
	}
	if p[0].Name != "Mu" {
		panic("normal: " + panicNameMismatch)
	}
	if p[1].Name != "Sigma" {
		panic("normal: " + panicNameMismatch)
	}
	n.Mu = p[0].Value
	n.Sigma = p[1].Value
}
// parameters returns the parameters of the distribution.
func (n Normal) parameters(p []Parameter) []Parameter {
	nParam := n.NumParameters()
	if p == nil {
		p = make([]Parameter, nParam)
	} else if len(p) != nParam {
		panic("normal: improper parameter length")
	}
	p[0].Name = "Mu"
	p[0].Value = n.Mu
	p[1].Name = "Sigma"
	p[1].Value = n.Sigma
	return p
}
```

Right now the breakdown is:

**Has:**
Normal
Exponential
Categorical (although by a different name)
Laplace (but called Mashal/UnmarshalParameters)
Triangle  (but called Mashal/UnmarshalParameters)
Uniform  (but called Mashal/UnmarshalParameters)
Weibull

**Doesn't Have:**
Poisson
Bernoulli
Beta
ChiSquared
F
Gamma
Gumbel
LogNormal
Pareto
StudentT

The motivation here for me is that I need an interface for distributions that has methods both for logprob and for setting parameters. I'm not completely sure what the 'ParameterMarshaler' interface alluded to is for - perhaps somebody else had a similar idea?  

Is this something that would be a useful contribution?

**Comment by btracey** [comment]
Yes, the `ParameterMarshaler` interface has exactly what you intended in mind. 

I'm fully convinced that the behavior you would like is useful. Needing LogProb and a way to set a parameter is exactly what I had in mind with those methods. I am not convinced that the current way we have it is the correct way (though clearly a partial solution is worse). The problem is that if you're trying to say, optimize over parameters (i.e. do a maximum likelihood fit), in many cases you want to work in a different space than the parameters typically live in. For instance, you always want to work with the log of the variance of the Normal distribution rather than the variance directly. It may be that having an additional type for doing this kind of thing (which could also include derivatives, a la ScoreParameters) would be better suited than expanding the MarshalParameter set.

**Comment by NMRobert** [comment]
Glad to hear we're in agreement on the utility! 

I think your concerns regarding the transformed parameter space are valid; after all, transforming parameters to unconstrained spaces occur pretty frequently. 

Perhaps the user provides an optional transformation function F: float64->float64 for each parameter such that the Get/Set calls essentially are mapping through it, and you'd probably also want to be able to provide a derivative for the transformation as well (as you noted) as we may need to scale the resulting probability density. Is this kinda what you have in mind? The default would just be an identity mapping (no-op).

I worry that this might be a bit counter-intuitive for new users and a bit clunky in general - it might not be the worst thing to just expose the Get/Set methods and leave it up to the user to deal with transformations as they require for their particular task. Golang does have a 'some assembly required' flavour anyways. What are your thoughts?
